File: frontend/src/services/authService.ts
import axios from "axios"
import api from "./api"
import type { LoginRequest, LoginResponse } from "@/types"

// Uses plain axios (not the api instance) to avoid the auth interceptor on these calls
const authApi = axios.create({
  baseURL: "/api/auth",
  headers: { "Content-Type": "application/json" },
})

export const authService = {
  login: async (data: LoginRequest): Promise<LoginResponse> => {
    const res = await authApi.post("/login", data)
    return res.data.data as LoginResponse
  },

  refresh: async (refreshToken: string): Promise<LoginResponse> => {
    const res = await authApi.post("/refresh", { refreshToken })
    return res.data.data as LoginResponse
  },

  logout: async (): Promise<void> => {
    const stored = localStorage.getItem("hrms-auth")
    const token = stored ? JSON.parse(stored)?.state?.accessToken : null
    if (token) {
      await authApi.post("/logout", null, {
        headers: { Authorization: `Bearer ${token}` },
      }).catch(() => {/* ignore logout errors */})
    }
  },

  changePassword: async (currentPassword: string, newPassword: string): Promise<void> => {
    await api.post("/auth/change-password", { currentPassword, newPassword })
  },
}

File: backend/src/main/java/com/hrms/auth/service/AuthService.java
// Auth service interface; implementations are swappable without touching controllers
package com.hrms.auth.service;

import com.hrms.auth.dto.ChangePasswordRequest;
import com.hrms.auth.dto.LoginRequest;
import com.hrms.auth.dto.LoginResponse;
import com.hrms.auth.dto.RefreshTokenRequest;

public interface AuthService {

    LoginResponse login(LoginRequest request);

    LoginResponse refreshToken(RefreshTokenRequest request);

    void logout(String username);

    void changePassword(String username, ChangePasswordRequest request);
}
